### PR TITLE
PROJQUAY-1079 - do not rescan bogus manifests

### PR DESCRIPTION
**Pull request by thomasmckay**
**Issue:** https://issues.redhat.com/browse/PROJQUAY-1079

**Changelog:** 

**Docs:** 

**Testing:** 
Confirm that when quay sends manifests to clair that it cannot parse, that the row in db is updated to -2 and not rescanned further.

**Details:** 
A helm repo was being pushed to clair with an empty layer array. Clair responds w/ a 400/BadRequest. Treat this now as an indication that clair _cannot_ scan this and should not retry. A further check was added to prevent empty layers from being sent to begin with.

### CODE REVIEW

File: data/secscan_model/secscan_v4_model.py [hunk 1]
```diff
@@ -190,6 +190,7 @@ def needs_reindexing_query(indexer_hash):
                 Manifest.select()
                 .join(ManifestSecurityStatus)
                 .where(
+                    ManifestSecurityStatus.index_status != IndexStatus.MANIFEST_UNSUPPORTED,
```
Comment (thomasmckay) [inline data/secscan_model/secscan_v4_model.py:193]: do not attempt to rescan unsupported manifests

File: data/secscan_model/secscan_v4_model.py [hunk 2]
```diff
@@ -211,10 +212,25 @@ def needs_reindexing_query(indexer_hash):
...
+                    indexer_hash="none",
+                    indexer_version=IndexerVersion.V4,
+                    metadata_json={},
+                )
+
         for candidate, abt, num_remaining in iterator:
             manifest = ManifestDataType.for_manifest(candidate, None)
             layers = registry_model.list_manifest_layers(manifest, self.storage, True)
-            if layers is None:
+            if layers is None or layers == []:
```
Comment (thomasmckay) [inline data/secscan_model/secscan_v4_model.py:233]: helm charts have an empty array of layers, which is not scannable

Comment (alecmerdler) [reply data/secscan_model/secscan_v4_model.py:233]: Maybe `len(layers) == 0` might be more readable.

File: data/secscan_model/secscan_v4_model.py [hunk 3]
```diff
@@ -211,10 +212,25 @@ def needs_reindexing_query(indexer_hash):
             ),
         )
 
+        def manifest_unsupported_status(manifest):
```
Comment (alecmerdler) [inline data/secscan_model/secscan_v4_model.py:215]: Perhaps rename to `mark_manifest_unsupported` to better represent the effect calling this function has.

File: data/secscan_model/secscan_v4_model.py (hunk 1 above)
Comment (alecmerdler) [reply data/secscan_model/secscan_v4_model.py:193]: Can we add a unit test for this?

**Review by alecmerdler — CHANGES_REQUESTED** [review]
Looks great, let's just add a unit test to prevent regressions.

**Review by alecmerdler — APPROVED** [review]

**Pull request merged**